pkg/execution/executor/executor.go

```diff
@@ -1712,14 +1718,15 @@ func (e *executor) Resume(ctx context.Context, pause state.Pause, r execution.Re
 			// Delete this pause, as an event has occured which matches
 			// the timeout.  We can do this prior to leasing a pause as it's the
 			// only work that needs to happen
-			err := e.pm.ConsumePause(ctx, pause.ID, nil)
+			_, _, err = e.pm.ConsumePause(ctx, pause.ID, nil)
 			if err == nil || err == state.ErrPauseNotFound {
 				return nil
 			}
 			return fmt.Errorf("error consuming pause via timeout: %w", err)
 		}
 
-		if err = e.pm.ConsumePause(ctx, pause.ID, r.With); err != nil {
+		didConsume, hasPendingSteps, err := e.pm.ConsumePause(ctx, pause.ID, r.With)
```

Comment:
This is the complex new return from the state store. We could replace this with a single `canContinue` bool instead of these individual vars, but it feels strange for the state store to be making that decision.

Reply:
Changed to a struct.